### DISCUSSION TITLE
PHP 8.4 Tracy does not show bluescreen with error: Deprecated: A::__construct(): Implicitly marking parameter $bar as nullable is deprecated, the explicit nullable type must be used instead

### DISCUSSION THREAD

**Discussion by petrparolek** [nette/tracy#590] [General]
Version:  2.10.9
PHP version: 8.4.1

### Bug Description
Tracy does not show bluescreen.

### Steps To Reproduce
```php
<?php

declare(strict_types = 1);

require __DIR__ . '/../vendor/autoload.php';

Tracy\Debugger::enable(\Tracy\Debugger::Development);

final class A
{

	private string $bar;

	public function __construct(string $bar = null) {
		$this->bar = $bar ?? 'foo';
	}

	public function getBar(): string
	{
		return $this->bar;
	}

}


$a = new A();
bdump($a->getBar());
```

### Expected Behavior
show bluescreen with error: `Deprecated: A::__construct(): Implicitly marking parameter $bar as nullable is deprecated, the explicit nullable type must be used instead in /var/www/html/www/tracy-test.php on line 14`

**Comment by dg** [comment]
You must first initialize Tracy, and only then can you load the file containing the error.

**Comment by mabar** [comment]
More specifically - PHP first processes all definitions (function, classes, traits, ...) and then rest of the code. In this case definition is loaded, then Tracy.

**Comment by petrparolek** [comment]
> You must first initialize Tracy, and only then can you load the file containing the error.

How does it work? Can you show me example code?
